sdks/python/apache_beam/testing/test_stream_test.py

```diff
@@ -360,24 +352,17 @@ def process(self,
         yield (elm, ts, side)
 
     records = (main_stream     # pylint: disable=unused-variable
-               | beam.ParDo(RecordFn(), beam.pvalue.AsList(side))
-               | beam.Map(recorded_elements))
+               | beam.ParDo(RecordFn(), beam.pvalue.AsList(side)))
     p.run()
 
-    # TODO(BEAM-3377): Remove after assert_that in streaming is fixed.
-    self.assertEqual([('a', Timestamp(2), [2]),
-                      ('b', Timestamp(4), [4])], result)
+    assert_that(records, equal_to(
```

Comment:
The `assert_that` modifies the graph. For it to actually work, it needs to be above `p.run()`. Otherwise it will not be tested. In this you can see that it should have failed if it was working, because it is checking for element`'e'`, however the main elements are actually `'a', 'b'`.

Reply:
Oops! That was a bad copy & paste from my dirty branch to the clean one. In any case, the goal I had was to see if the format was OK, so I could then move things to the assert_that itself (which I have just done).